nixos/modules/services/home-automation/zwave-js.nix

```
@@ -0,0 +1,158 @@
+{config, pkgs, lib, ...}:
+
+with lib;
+
+let
+  cfg = config.services.zwave-js;
+  cacheDir = "/var/cache/zwave-js";
+  runtimeDir = "/run/zwave-js";
+  defaultConfig = {
+    storage = {
+      inherit cacheDir;
+    };
+  };
+  mergedConfigFile = "${runtimeDir}/config.json";
+in {
+
+  options.services.zwave-js = {
+    enable = mkEnableOption (mdDoc "Start the zwave-js server on boot");
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.zwave-js-server;
+      defaultText = literalExpression "pkgs.zwave-js-server";
+      description = mdDoc ''
+        Package for zwave-js server
+      '';
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 3000;
+      description = mdDoc ''
+        Port for the server to listen on.
+      '';
+    };
+
+    serialPort = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = mdDoc ''
+        Serial port device path for Z-Wave controller.
+      '';
+      example = "/dev/ttyUSB0";
+    };
+
+    secretsConfigFile = mkOption {
+      type = types.path;
+      description = mdDoc ''
+        JSON file containing secret keys. This file will be merged with the
+        module-generated config file (taking precedence).
+
+        Z-Wave keys can be generated with:
+
+          {command}`< /dev/urandom tr -dc A-F0-9 | head -c32 ;echo`
+
+        ::: {.warning}
+        A file in the nix store should not be used since it will be readable to
+        all users.
+        :::
+      '';
+      example = ''
+        {
+          "securityKeys": {
+            "S2_Unauthenticated": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
+            "S2_Authenticated": "BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB",
+            "S2_AccessControl": "CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC",
+            "S0_Legacy": "DDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD"
+          }
+        }
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = types.attrs;
+      default = {};
+      description = mdDoc ''
+        Extra configuration to be merged into the generated config file.
+      '';
+      example = literalExpression ''
+        {
+          storage = {
+            throttle = "fast";
+          };
+        }
+      '';
+    };
+
+    mockDriver = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc ''
+        Use a mock USB serial driver (mutually exclusive with serialPort option).
+      '';
+      internal = true;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.mockDriver && cfg.serialPort == null || !cfg.mockDriver && cfg.serialPort != null;
+        message = "Set either serialPort or mockDriver but not both";
```

```suggestion
        message = "The zwave-js-server module requires one of serialPort or mockDriver but not both";
```

It's a good practice to mention the module, where the assertion message originates from. Anything like that could work.